File: Cpp/1299_Replace_Elements_with_Greatest_Element_on_Right_Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        vector<int> result(arr.size());
        int greatest_element = arr.back();
        
        result[arr.size()-1] = -1;
        for(int i=arr.size()-2; i>=0; --i){
            result[i] = greatest_element;
            if(greatest_element < arr[i]){
                greatest_element = arr[i];
            }
        }
        
        return result;
    }
};

File: Kotlin/1299_Replace_Elements_with_Greatest_Element_on_Right_Side.kt
class Solution {
    fun replaceElements(arr: IntArray): IntArray {
        val result = IntArray(arr.size)
        var greatestElement = arr.last()
        result[arr.size - 1] = -1
        
        for(i in arr.size - 2 downTo 0){
            result[i] = greatestElement
            
            if(greatestElement < arr[i]){
                greatestElement = arr[i]
            }
        }
        
        return result
    }
}